File: Data Structures & Algorithms/two-integer-sum/submission-0.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int[] pair = new int[2];
        for(int i=0; i<nums.length-1;i++){
            for(int j=nums.length-1; j>i; j--){
                if(nums[i]+nums[j] == target){
                    pair[0] = i;
                    pair[1] = j;
                }
            }
        }
        return pair;
    }
}

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        n = len(nums)
        res = [1] * n  # This space is required for the output, so it's O(1) extra space.
        
        # Loop 1: O(N) Time
        left_val = 1
        for i in range(n):
            res[i] = left_val
            left_val *= nums[i]
            
        # Loop 2: O(N) Time
        right_val = 1
        for i in range(n - 1, -1, -1):
            res[i] *= right_val
            right_val *= nums[i]
            
        return res